### PR TITLE
feat(DDS): add dds backup resource

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add dds backup resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add dds backup resource
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dds/' TESTARGS='-run TestAccDdsBackup_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dds/ -v -run TestAccDdsBackup_basic -timeout 360m -parallel 4
=== RUN   TestAccDdsBackup_basic
=== PAUSE TestAccDdsBackup_basic
=== CONT  TestAccDdsBackup_basic
--- PASS: TestAccDdsBackup_basic (935.48s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dds       935.524s
```

### CODE REVIEW

File: docs/resources/dds_backup.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+## Timeouts
+
+This resource provides the following timeouts configuration options:
+
+* `create` - Default is 30 minutes.
+* `delete` - Default is 10 minutes.
+
+## Import
+
+The dds backup can be imported using the `id`, e.g.
```
Comment (Jason-Zhang9309) [inline docs/resources/dds_backup.md]: looks instance_id is also needed to import the backup

File: huaweicloud/services/dds/resource_huaweicloud_dds_backup.go [hunk 2]
```diff
@@ -0,0 +1,535 @@
...
+	jobId, err := jmespath.Search("job_id", deleteBackupRespBody)
+	if err != nil {
+		return diag.Errorf("error deleting DdsBackup: job_id is not found in API response")
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"Running"},
+		Target:       []string{"Deleted"},
+		Refresh:      ddsJobStatusRefreshFunc(jobId.(string), region, cfg),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/dds/resource_huaweicloud_dds_backup.go]:
```suggestion
		Timeout:      d.Timeout(schema.TimeoutDelete),
```

File: docs/resources/dds_backup.md (hunk 1 above)
Comment (houpeng80) [reply docs/resources/dds_backup.md]: done

File: huaweicloud/services/dds/resource_huaweicloud_dds_backup.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/dds/resource_huaweicloud_dds_backup.go]: done

File: huaweicloud/services/dds/resource_huaweicloud_dds_backup.go [hunk 3]
```diff
@@ -0,0 +1,550 @@
...
+			"datastore": {
+				Type:        schema.TypeList,
+				Elem:        BackupDatastoreSchema(),
+				Computed:    true,
+				Description: `Indicates the database version.`,
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the backup type. `,
```
Comment (ShiChangkuo) [inline huaweicloud/services/dds/resource_huaweicloud_dds_backup.go]:
```suggestion
				Description: `Indicates the backup type.`,
```
remove the extra space

File: huaweicloud/services/dds/resource_huaweicloud_dds_backup.go [hunk 4]
```diff
@@ -0,0 +1,550 @@
...
+	}
+
+	createBackupRespBody, err := utils.FlattenResponse(createBackupResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := jmespath.Search("backup_id", createBackupRespBody)
+	if err != nil {
+		return diag.Errorf("error creating DdsBackup: ID is not found in API response")
```
Comment (ShiChangkuo) [inline huaweicloud/services/dds/resource_huaweicloud_dds_backup.go]:
```suggestion
		return diag.Errorf("error creating DDS backup: ID is not found in API response")
```

please update it in all files

File: huaweicloud/services/dds/resource_huaweicloud_dds_backup.go [hunk 5]
```diff
@@ -0,0 +1,550 @@
...
+		return common.CheckDeletedDiag(d, err, "error retrieving DdsBackup")
+	}
+
+	getBackupRespBody, err := utils.FlattenResponse(getBackupResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	backups := utils.PathSearch("backups", getBackupRespBody, make([]interface{}, 0)).([]interface{})
+	if len(backups) == 0 {
+		return diag.Errorf("error get backup by backup ID %s", d.Id())
```
Comment (ShiChangkuo) [inline huaweicloud/services/dds/resource_huaweicloud_dds_backup.go]: should call CheckDeleted

Comment (houpeng80) [reply huaweicloud/services/dds/resource_huaweicloud_dds_backup.go]: done

File: huaweicloud/services/dds/resource_huaweicloud_dds_backup.go (hunk 4 above)
Comment (houpeng80) [reply huaweicloud/services/dds/resource_huaweicloud_dds_backup.go]: done

File: huaweicloud/services/dds/resource_huaweicloud_dds_backup.go (hunk 3 above)
Comment (houpeng80) [reply huaweicloud/services/dds/resource_huaweicloud_dds_backup.go]: done

File: huaweicloud/services/dds/resource_huaweicloud_dds_backup.go [hunk 6]
```diff
@@ -0,0 +1,549 @@
...
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        60 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for instance (%s) to become ready: %s", id.(string), err)
+	}
+	d.SetId(id.(string))
```
Comment (ShiChangkuo) [inline huaweicloud/services/dds/resource_huaweicloud_dds_backup.go]: please set the resource id immediately we get the ID from API response

**Pull request merged**